src/common/app_client.rs

```diff
@@ -172,6 +172,59 @@ impl<'a> AppClientBuilder<'a> {
     }
 }
 
+pub fn fetch_config(
```

Comment:
there is a lot of code duplication here what was the rational behind it?

Reply:
I wanted to be able to do this outside of having to create an AppClientBuilder because it felt strange to create one of those but never construct an AppClient

Reply:
I see, I think it's fin to build an app client just to take ownership of the config and drop it afterward. then we can have the robot constructor be name new_from_config.

Reply:
is [this](https://github.com/viamrobotics/micro-rdk/pull/50/commits/fac3f7161209cf69b8a422e0977efa4da68a8ec5) what you had in mind?